exporter/datadogexporter/config/config.go

```diff
@@ -104,6 +104,14 @@ type TracesConfig struct {
 	// all entries must be surrounded by double quotes and separated by commas.
 	// ignore_resources: ["(GET|POST) /healthcheck"]
 	IgnoreResources []string `mapstructure:"ignore_resources"`
+
+	// SpanNameRemappings is the list of datadog span names and preferred names to map to. This can be used to
+	// automatically map Datadog Span Operation Names to an updated value, and is useful when a user wants to
+	// shorten or modify span names to something more user friendly in the case of instrumentation libraries with
+	// particularly verbose names. All entries must be surrounded by double quotes and be separated by commas, with each entry
+	// between old and updated name separated by a space.
+	// span_name_remappings: ["io.opentelemetry.javaagent.spring.client spring.client", "instrumentation::express.server express"]
+	SpanNameRemappings []string `mapstructure:"span_name_remappings"`
```

Comment:
I guess not, but, is it not possible to do this on a processor? It feels like this is not specific to us. I am going to keep on reviewing assuming it is not but I think we should answer this

Reply:
Oddly enough I couldn't find a processor that allows us to modify instrumentation library name fields.

Reply:
Can you have a look and see if we can add it to an existing component? I think we should avoid adding it here if it's something that would be useful for other exporters

Reply:
the functionality does not exist within a current processor. I think there's some ability to filter out spans from specific ILname+ILversions, but nothing that allows mutation of those values: confirmed with maintainers on slack 
![image](https://user-images.githubusercontent.com/14250318/119565685-4073fb00-bd78-11eb-8a3a-95d437b1a6d7.png)

Reply:
So, to clarify: I understand that there is currently no processor that does this and that this is something that users would find useful. What I meant to say is that instead of adding this functionality to the Datadog exporter it might be better to add it to a processor, since it seems like this functionality could be leveraged by anyone.

If it's too hard to do that right now then we can consider adding this in some sort of experimental/unstable way to our exporter and move it later to a processor, but I would like to avoid adding functionality that is not specific to Datadog here.

Reply:
it's specific to datadog in that the datadog span operation name is a concatenation of 2 different opentelemetry span related fields (IL_Name + Span_Kind), and has critical significance to the way trace metrics are generated and displayed in the datadog ui.

additionally, a user may use the un-mutated IL_Name and the un-mutated Span_kind for other purposes within both our exporter and within other vendor/oss exporters, and so even if we did the work to add this functionality to a processor, polluting those names by modifying them upstream would be problematic. At the very least, it would force users to run a separate pipeline just for exporting to datadog, and even then, it would require users to learn the internals of how these span names are generated so that they could determine what fields to modify. It's just simpler to have this in our exporter imo.

On top of that, practically speaking it is something users want and anecdotally it seems like any attempts to add new features to processors in `core` is completely blocked right now since the focus is on 1.0. If we have to add a `contrib` processor here, then we wouldn't same guarantees that this processor would be included in cloud vendor distros like ADOT which our users may be running.

I'm fine with this being marked experimental for now if that helps this make it into the exporter

Reply:
> it's specific to datadog in that the datadog span operation name is a concatenation of 2 different opentelemetry span related fields (IL_Name + Span_Kind),

Okay, that is convincing to me, sorry, I didn't have enough context on this and I thought it was more general. It looks good to me then, thanks for bearing with me :)